internal/datastore/common/sql.go

```diff
@@ -467,10 +467,11 @@ func (sqf SchemaQueryFilterer) FilterWithRelationshipsFilter(filter datastore.Re
 		csqf = csqf.FilterWithCaveatName(filter.OptionalCaveatName)
 	}
 
-	if filter.OptionalExpirationOption == datastore.ExpirationFilterOptionHasExpiration {
+	switch filter.OptionalExpirationOption {
```

Comment:
This was one of a set of new lint issues that were automatically fixed by --fix.